### DISCUSSION TITLE
How to determine which GitHub REST API operations aren't in rest.js

### DISCUSSION THREAD

**Discussion by rachmari** [octokit/octokit.js#2032] [Q&A] [answered]
👋 @gr2m 

I had a couple of questions about the GH Enterprise REST API. It looks like rest.js doesn't support the API operations for enterprise administration and management console, is that correct?
- https://docs.github.com/en/enterprise-server@3.0/rest/reference/enterprise-admin

If not, it makes sense but I'm curious if we should describe this in the OpenAPI schema in some way. For example, we shouldn't automatically generate an Octokit example if the library doesn't support the operation:
- https://docs.github.com/en/enterprise-server@3.0/rest/reference/enterprise-admin#add-an-authorized-ssh-key

Or perhaps we're better off having an exclude list that we check against directly in the https://github.com/github/docs-internal repo?

**Comment by wolfy1339** [comment]
There is this [plugin](https://github.com/octokit/plugin-enterprise-server.js) to use with `@octokit/rest` that adds the enterprise APIs.
In it's repo, there is this README that contains information on Enterprise admin routes: https://github.com/octokit/plugin-enterprise-server.js/blob/master/docs/ghe-222.md#admin

**Comment by gr2m** [comment, marked as answer]
The examples in the docs are not using `@octokit/rest`, but [`@octokit/core`](https://github.com/octokit/core.js). All current and future endpoints are automatically supported by `@octokit/core`, it's just a slight wrapper around a generic request method.

`@octokit/rest` adds methods for each endpoint, which is just for convenience. Because it adds quite a lot of code, `@octokit/rest` does not include the enterprise methods. Instead, you can compose your own Octokit with `@octokit/core` and then the plugins you need:

- github.com/octokit/plugin-rest-endpoint-methods.js (which `@octokit/rest` is using, see composition [here](https://github.com/octokit/rest.js/blob/54784482b69ff4929c405aa06008869c950ec3a1/src/index.ts))
- github.com/octokit/plugin-enterprise-server.js
- github.com/octokit/plugin-enterprise-cloud.js

so tl;dr, I think we are good
